classes/01class/exercises/c01-aws02/rodrigovcesar/README.md

```diff
@@ -0,0 +1,147 @@
+# C01-AWS02
+
+## Commands Execution Output
+
+- Commands to create the S3 Bucket and to copy the file to the bucket:
+```
+aws s3 mb s3://devopsacademy-rodrigo --region ap-southeast-2
+make_bucket: devopsacademy-rodrigo
+
+touch file_for_s3.txt
+rodrigo@rodrigo-Aspire-V5-471:~/DevOps/academy/classes/01class/exercises/c01-aws02/rodrigovcesar$ echo 'Content of the file for test' > file_for_s3.txt
+
+aws s3 cp ./file_for_s3.txt s3://devopsacademy-rodrigo/
+upload: ./file_for_s3.txt to s3://devopsacademy-rodrigo/file_for_s3.txt
+
+```
+
+- Commands to allow the EC2 instance to access the files in S3:
+```
+cat role-ec2.json 
+{
+  "Version": "2012-10-17",
+  "Statement": {
+    "Effect": "Allow",
+    "Principal": {"Service": "ec2.amazonaws.com"},
+    "Action": "sts:AssumeRole"
+  }
+}
+
+
+aws iam create-role --role-name EC2S3Role --assume-role-policy-document=file://role-ec2.json
+{
+    "Role": {
+        "Path": "/",
+        "RoleName": "EC2S3Role",
+        "RoleId": "AROARP4GZM4KY3HSV6OWC",
+        "Arn": "arn:aws:iam::102825027349:role/EC2S3Role",
+        "CreateDate": "2020-06-20T11:08:04+00:00",
+        "AssumeRolePolicyDocument": {
+            "Version": "2012-10-17",
+            "Statement": {
+                "Effect": "Allow",
+                "Principal": {
+                    "Service": "ec2.amazonaws.com"
+                },
+                "Action": "sts:AssumeRole"
+            }
+        }
+    }
+}
+
+cat restrictedS3Policy.json 
+{
+  "Version": "2012-10-17",
+  "Statement": {
+    "Effect": "Allow",
+    "Action": "s3:ListBucket",
```

Comment:
this role does not have s3:GetObject to `arn:aws:s3:::devopsacademy-rodrigo/*` so using this role you won't be able to get any objects from the bucket. I guess that's what you have mentioned on the challenges faced?

Reply:
Sorry, I forgot to make this change here, just made another commit now. Partially about the challenges faced, although I was referring more about trying to connect to S3 from the private instance.